### PR TITLE
chore: reduce build time by converting docker image to build with nix

### PR DESCRIPTION
**Pull request by kylecarbs**
The build time goes from ~40m to ~1m and aligns us with our Nix users.

### CODE REVIEW

**Comment by matifali** [comment]
Will this also have docker in docker enabled?

**Comment by kylecarbs** [comment]
@matifali it does indeed!

**Review by code-asher — APPROVED** [review]
Very cool!

File: .github/workflows/dogfood.yaml [hunk 1]
```diff
@@ -5,19 +5,26 @@ on:
     branches:
       - main
     paths:
+      - "flake.nix"
+      - "flake.lock"
       - "dogfood/**"
       - ".github/workflows/dogfood.yaml"
   # Uncomment these lines when testing with CI.
```
Comment (aslilac) [inline .github/workflows/dogfood.yaml:12]: comment might be out of date?

File: flake.nix [hunk 2]
```diff
@@ -11,50 +11,186 @@
...
+            cp ${pkgs.sudo}/bin/sudo usr/bin/sudo
+            chmod 4755 usr/bin/sudo
+            mkdir -p /etc/init.d
+          '';
+        };
+        # Environment variables that live in `/etc/environment` in the container.
+        # These will also be applied to the container config.
+        devEnvVars = [
+          "PATH=${pkgs.lib.makeBinPath devShellPackages}:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/home/coder/go/bin"
+          #This setting prevents Go from using the public checksum database for
```
Comment (aslilac) [inline flake.nix:101]:
```suggestion
          # This setting prevents Go from using the public checksum database for
```

File: .github/workflows/dogfood.yaml (hunk 1 above)
Comment (kylecarbs) [reply .github/workflows/dogfood.yaml:12]: Ahh, I almost missed this ty!

**Pull request merged**